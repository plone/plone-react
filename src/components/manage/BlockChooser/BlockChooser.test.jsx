import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import BlockChooser from './BlockChooser';

const mockStore = configureStore();

test('renders a BlockChooser component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <BlockChooser onMutateBlock={() => {}} currentBlock="theblockid" />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
