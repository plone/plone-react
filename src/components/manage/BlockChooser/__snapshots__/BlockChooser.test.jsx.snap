// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlocksChooser Fallback BlockChooser component onMutateBlock 1`] = `
<div>
  <div
    class="blocks-chooser"
  >
    <div
      class="accordion ui fluid styled form"
    >
      <div
        aria-label="Fold Most used blocks"
        class="active title"
      >
        Most used
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content active mostUsed"
      >
        <div
          aria-hidden="false"
          class="rah-static rah-static--height-auto"
          style="height: auto; overflow: visible;"
        >
          <div
            style="transition: opacity 500ms ease 0ms;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button image"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Image
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button listing"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Listing
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button video"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Video
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Unfold Text blocks"
        class="title"
      >
        Text
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content text"
      >
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 500ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button text"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Text
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Unfold Media blocks"
        class="title"
      >
        Media
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content media"
      >
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 500ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button image"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Image
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button leadimage"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Lead Image Field
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button video"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Video
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Unfold Common blocks"
        class="title"
      >
        Common
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content common"
      >
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 500ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button listing"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Listing
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button toc"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Table of Contents
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button hero"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Hero
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button maps"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Maps
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button html"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  HTML
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button table"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Table
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BlocksChooser renders a BlockChooser component 1`] = `
<div>
  <div
    class="blocks-chooser"
  >
    <div
      class="accordion ui fluid styled form"
    >
      <div
        aria-label="Fold Most used blocks"
        class="active title"
      >
        Most used
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content active mostUsed"
      >
        <div
          aria-hidden="false"
          class="rah-static rah-static--height-auto"
          style="height: auto; overflow: visible;"
        >
          <div
            style="transition: opacity 500ms ease 0ms;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button image"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Image
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button listing"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Listing
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button video"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Video
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Unfold Text blocks"
        class="title"
      >
        Text
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content text"
      >
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 500ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button text"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Text
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Unfold Media blocks"
        class="title"
      >
        Media
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content media"
      >
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 500ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button image"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Image
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button leadimage"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Lead Image Field
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button video"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Video
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Unfold Common blocks"
        class="title"
      >
        Common
        <div
          class="accordion-tools"
        >
          <svg
            class="icon"
            style="height: 20px; width: auto; fill: currentColor;"
            viewBox=""
            xmlns=""
          />
        </div>
      </div>
      <div
        class="content common"
      >
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 500ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button listing"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Listing
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button toc"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Table of Contents
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button hero"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Hero
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button maps"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Maps
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button html"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  HTML
                </span>
              </button>
            </div>
            <div
              class="ui buttons"
            >
              <button
                class="ui basic icon button table"
              >
                <svg
                  class="icon"
                  style="height: 36px; width: auto; fill: currentColor;"
                />
                <span>
                  Table
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
